pkg/playback/http_playback.go

```diff
@@ -0,0 +1,413 @@
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		err = rr.setCassetteDir(absoluteCassetteDir)
+
+		if err != nil {
+			fmt.Println("Error in /playback/cassette/setroot handler: ", err)
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+		// TODO: why is this not printing out the absolute path?
+		fmt.Printf("Cassette directory set to \"%v\"\n", rr.cassetteDirectory)
+		w.WriteHeader(200)
+	})
+
+	customMux.HandleFunc("/playback/cassette/load", func(w http.ResponseWriter, r *http.Request) {
+		filepathVals, ok := r.URL.Query()["filepath"]
+
+		if !ok {
+			err := fmt.Errorf("\"filepath\" query param must be present")
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		if len(filepathVals) > 1 {
+			fmt.Printf("Multiple \"filepath\" param values given, ignoring all except first: %v\n", filepathVals[0])
+		}
+
+		relativeFilepath := filepathVals[0]
+
+		if !strings.HasSuffix(strings.ToLower(relativeFilepath), ".yaml") {
+			err := fmt.Errorf("%v is not a .yaml file", relativeFilepath)
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		if filepath.IsAbs(relativeFilepath) {
+			err := fmt.Errorf("%v must be a relative filepath. a absolute filepath was provided", relativeFilepath)
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		err := rr.loadCassette(relativeFilepath)
+
+		if err != nil {
+			writeErrorToHTTPResponse(w, err, 500)
+			return
+		}
+
+		var isRecording bool
+		switch rr.mode {
+		case Record:
+			isRecording = true
+		case Replay:
+			isRecording = false
+		case Auto:
+			isRecording = rr.isRecordingInAutoMode
+		default:
+			writeErrorToHTTPResponse(w, errors.New("in unexpected mode state in handler. Please restart the playback server"), 500)
+			return
+		}
+
+		var statusMsg string
+
+		if isRecording {
+			statusMsg = fmt.Sprintf("Recording to %v", relativeFilepath)
+		} else {
+			statusMsg = fmt.Sprintf("Replaying from %v", relativeFilepath)
+		}
+		fmt.Println(statusMsg)
+		w.WriteHeader(200)
+	})
+
+	customMux.HandleFunc("/playback/cassette/eject", func(w http.ResponseWriter, r *http.Request) {
+		err := rr.ejectCassette()
+
+		if err != nil {
+			writeErrorToHTTPResponse(w, err, 500)
+		} else {
+			w.WriteHeader(200)
+			fmt.Println("/playback/cassette/eject: Ejected cassette")
+		}
+		fmt.Println("")
+		fmt.Println("=======")
+		fmt.Println("")
+	})
+
+	// Display error message for unmatched routes with a control-prefix
+	customMux.HandleFunc("/playback/", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(404)
+		fmt.Fprintf(w, "\"%v\" is not a valid /playback/ control endpoint. Run `stripe playback --help` for a comprehensive list.\n", r.URL)
+	})
+
+	// --- Default handler
+	customMux.HandleFunc("/", rr.handler)
+
+	return server
+}
+
+func (rr *Server) handler(w http.ResponseWriter, r *http.Request) {
+	if !rr.cassetteLoaded {
+		w.WriteHeader(400)
+		fmt.Fprint(w, "No cassette is loaded.\n")
+		return
+	}
+
+	switch rr.mode {
```

Comment:
(non-blocking suggestion): 

You could DRY this up a little:
```go
if (rr.isRecording()) {
  ...
} else {
  ...
}
```
if you implemented
```go
func (rr *Server) isRecording() {
  switch rr.mode {
  case Record:
    return true
  case Replay:
    return false
  case Auto:
    return rr.isRecordingInAutoMode
  }
  default: //throw
}
```

Reply:
Done in 961d13ecaf1f0c8830e80fe070101dbe0ce126ab